### PR TITLE
Finished draft of L2 Construct with Rule Body as a JSON string. Also …

### PR DESCRIPTION
**Pull request by abrahcoh**
…added unit tests to confirm basic functionality. Later adding RuleName validation, RuleBody validation, and integration tests

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-cloudwatch/lib/insight-rule.ts [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+     *
+     * @param scope The parent creating construct (usually 'this')
+     * @param id The construct's name
+     * @param insightRuleName The Contributor Insights Rule's name
+     */
+  public static fromInsightRuleName(scope: Construct, id: string,
+    insightRuleName: string): IInsightRule {
+    class Import extends InsightRuleBase {
+      public readonly insightRuleArn = core.Stack.of(scope).formatArn({
+        service: 'cloudformation',
```
Comment (mickrile) [inline packages/@aws-cdk/aws-cloudwatch/lib/insight-rule.ts]: Should the service be cloudwatch?

File: packages/@aws-cdk/aws-cloudwatch/lib/insight-rule.ts [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+      ruleBody: this.ruleBody,
+      ruleState: this.ruleState || InsightRuleStates.ENABLED,
+    });
+
+    /**
+         * The arn and name are both attributes of the CFN resource
+         * @see https://docs.aws.amazon.com/cdk/api/latest/python/aws_cdk.aws_cloudwatch/CfnInsightRule.html
+         */
+    this.insightRuleArn = insightRule.attrArn;
+    this.insightRuleRuleName = insightRule.ref;
```
Comment (mickrile) [inline packages/@aws-cdk/aws-cloudwatch/lib/insight-rule.ts]: Could you explain why this is insightRule.ref instead insightRule.attr_rule_name?

File: packages/@aws-cdk/aws-cloudwatch/test/test.insights-rule.ts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+      RuleState: 'ENABLED',
+    }));
+
+    test.done();
+  },
+
+  'In InsightRules, Testing that all parameters exist with rule' (test: Test) {
+    //WHEN
+    const stack = new Stack();
+    new InsightRule(stack, 'rule2', {
```
Comment (mickrile) [inline packages/@aws-cdk/aws-cloudwatch/test/test.insights-rule.ts:76]: rule3?

File: packages/@aws-cdk/aws-cloudwatch/lib/insight-rule.ts (hunk 1 above)
Comment (abrahcoh) [reply packages/@aws-cdk/aws-cloudwatch/lib/insight-rule.ts]: Yes you are correct

File: packages/@aws-cdk/aws-cloudwatch/lib/insight-rule.ts (hunk 2 above)
Comment (abrahcoh) [reply packages/@aws-cdk/aws-cloudwatch/lib/insight-rule.ts]: I did not notice that insight rules had an attr rule name. All other constructs use the .ref; however, insights rule is unique to most of the other constructs, as they have this attribute. I'll use "attr_rule_name" for now, as it seems to be the equivalent and is more readable

**Review by mickrile — APPROVED** [review]

**Pull request merged**